ui/src/features/Dialog/components/workspace/AddWorkspace.tsx

```diff
@@ -0,0 +1,98 @@
+import { useNavigate } from "@tanstack/react-router";
+import { useEffect, useState } from "react";
+
+import { Button, Input } from "@flow/components";
+import {
+  CreateWorkspaceMutation,
+  useCreateWorkspaceMutation,
+  useGetWorkspaceQuery,
+} from "@flow/lib/gql";
+import { useT } from "@flow/lib/i18n";
+import { useDialogType } from "@flow/stores";
+
+import { ContentHeader } from "../ContentHeader";
+import { ContentSection } from "../ContentSection";
+
+const AddWorkspace: React.FC = () => {
+  const t = useT();
+  const [name, setName] = useState<string | undefined>();
+  const [buttonDisabled, setButtonDisabled] = useState(false);
+  const [, setDialogType] = useDialogType();
+  const navigate = useNavigate();
+  const [showError, setShowError] = useState(false);
+  const [createdWorkspaceId, setCreatedWorkspaceId] = useState<string | undefined>(undefined);
```

Comment:
Utilizing this adds a lot of code that isn't really necessary. Instead of setting this state, just to handle a navigation in a useEffect, I think its simpler and better to just await the mutation inside the handleClick function, and then redirect if there is a workspace.id.

Reply:
You can wait for the mutation but the query automatically refreshes after that and I need to wait for it to finish before navigating. Is there a better way in tanstack?

Reply:
I'll see if I can do better

Reply:
We have global state for currentWorkspace, so for starters could just await, and then when we get results we assign it to currentWorkspace state and, navigate.

Reply:
I figured it out. Need to await the refetch. 
Also optimistic cache or UI update won't work in this scenario because:
- We need the ID of the workspace 
- ID will be used to navigate 

We can only get the ID from the backend. I'll keep the optimistic UI and cache in mind for future.